plugins/javaembed/javaembed.cpp

```diff
@@ -1153,6 +1157,272 @@ class JavaRowStream : public CInterfaceOf<IRowStream>
+        objToString = JNIenv->GetMethodID(langObjectClass, "toString", "()Ljava/lang/String;");
+    }
+    ~JavaObjectXmlWriter()
+    {
+        if (Class)
+            JNIenv->DeleteGlobalRef(Class);
+        HashIterator it(javaClasses);
+        ForEach(it)
+        {
+            IMapping &entry = it.query();
+            jclass *pClass = javaClasses.mapToValue(&entry);
+            if (pClass)
+                JNIenv->DeleteGlobalRef(*pClass);
+        }
+    }
+
+    void writeSimpleType(jclass parentClass, jobject parentObject, IEsdlDefObject &defObject)
+    {
+        const char *fieldname = defObject.queryName();
+        const char *javaSig = esdl2JavaSig(esdl, defObject.queryProp("type"));
+        jfieldID fieldId = JNIenv->GetFieldID(parentClass, fieldname, javaSig); //tbd cache this
+        if (!fieldId)
+            return;
+        jobject fieldObj = (jobject) JNIenv->GetObjectField(parentObject, fieldId);
+        if (!fieldObj)
+            return;
+        jstring fieldStr = (jstring) JNIenv->CallObjectMethod(fieldObj, objToString);
+        const char *text = JNIenv->GetStringUTFChars(fieldStr, NULL);
+        if (!text)
+            return;
+        writer.outputCString(text, defObject.queryName());
+    }
+    void writeEnumType(jclass parentClass, jobject parentObject, IEsdlDefObject &defObject)
+    {
+        const char *fieldname = defObject.queryName();
+        VStringBuffer javaSig("L%s/%s;", esdlService.str(), defObject.queryProp("enum_type"));
+        jfieldID fieldId = JNIenv->GetFieldID(parentClass, fieldname, javaSig);
+        if (!fieldId)
+            return;
+        jobject fieldObj = (jobject) JNIenv->GetObjectField(parentObject, fieldId);
+        if (!fieldObj)
+            return;
+        jstring fieldStr = (jstring) JNIenv->CallObjectMethod(fieldObj, objToString);
+        const char *text = JNIenv->GetStringUTFChars(fieldStr, NULL);
+        if (!text)
+            return;
+        writer.outputCString(text, defObject.queryName());
+    }
+    void writeComplexType(jclass parentClass, jobject parentObject, IEsdlDefObject &defObject)
+    {
+        IEsdlDefStruct *defStruct = esdl.queryStruct(defObject.queryProp("complex_type"));
+        if (!defStruct)
+            return;
+        const char *fieldname = defObject.queryName();
+        VStringBuffer javaSig("L%s/%s;", esdlService.str(), defObject.queryProp("complex_type"));
+        jfieldID fieldId = JNIenv->GetFieldID(parentClass, fieldname, javaSig); //tbd cache this
+        if (!fieldId)
+            return;
+        jobject fieldObj = (jobject) JNIenv->GetObjectField(parentObject, fieldId);
+        if (!fieldObj)
+            return;
+        writer.outputBeginNested(fieldname, true);
+        writeChildren(JNIenv->GetObjectClass(fieldObj), fieldObj, defStruct);
+        writer.outputEndNested(fieldname);
+    }
+    void writeSimpleArray(jclass parentClass, jobject parentObject, IEsdlDefObject &defObject)
+    {
+        //tbd
```

Comment:
Is there a Jira open to make sure you revisit this? Should it say UNIMPLEMENTED; ?